packages/compass-schema/src/stores/schema-export-reducer.ts

```diff
@@ -319,3 +367,110 @@ export const schemaExportReducer: Reducer<SchemaExportState, Action> = (
 
   return state;
 };
+
+//// TODO: add ticket number to clean this up
+export type openLegacyBannerAction = {
+  type: SchemaExportActions.openLegacyBanner;
+};
+
+export const openLegacyBanner = (): SchemaThunkAction<void> => {
+  return (dispatch, getState) => {
+    const choiceInState = getState().schemaExport.legacyBannerChoice;
+    const savedChoice = choiceInState || localStorage.getItem(localStorageId);
+    if (savedChoice) {
+      if (savedChoice !== choiceInState) {
+        dispatch({
+          type: SchemaExportActions.setLegacyBannerChoice,
+          choice: savedChoice,
+        });
+      }
+      if (savedChoice === 'legacy') {
+        dispatch(confirmedLegacySchemaShare());
+        return;
+      }
+      if (savedChoice === 'export') {
+        dispatch(openExportSchema());
+        return;
+      }
+    }
+    dispatch({ type: SchemaExportActions.openLegacyBanner });
+  };
+};
+
+export type closeLegacyBannerAction = {
+  type: SchemaExportActions.closeLegacyBanner;
+};
+
+export type setLegacyBannerChoiceAction = {
+  type: SchemaExportActions.setLegacyBannerChoice;
+  choice: 'legacy' | 'export';
+};
+
+const localStorageId = 'schemaExportLegacyBannerChoice';
+
+export const switchToSchemaExport = (): SchemaThunkAction<void> => {
+  return (dispatch) => {
+    dispatch({ type: SchemaExportActions.closeLegacyBanner });
+    dispatch(openExportSchema());
+  };
+};
+
+export const confirmedLegacySchemaShare = (): SchemaThunkAction<void> => {
+  return (dispatch, getState, { namespace }) => {
+    const {
+      schemaAnalysis: { schema },
+    } = getState();
+    const hasSchema = schema !== null;
+    if (hasSchema) {
+      void navigator.clipboard.writeText(JSON.stringify(schema, null, '  '));
+    }
+    dispatch(_trackSchemaShared(hasSchema));
+    dispatch({ type: SchemaExportActions.closeLegacyBanner });
+    openToast(
+      'share-schema',
+      hasSchema
+        ? {
+            variant: 'success',
+            title: 'Schema Copied',
+            description: `The schema definition of ${namespace} has been copied to your clipboard in JSON format.`,
+            timeout: 5_000,
+          }
+        : {
+            variant: 'warning',
+            title: 'Analyze Schema First',
+            description: 'Please Analyze the Schema First from the Schema Tab.',
+            timeout: 5_000,
```

Comment:
This already existed, but I figure we can improve it as we're in the code now, feel free not to action. The timeout here we could remove as its something the user should have time to read and action on. I also find wording is a bit odd in its choice of capitalization. The other toasts we have in Compass don't have capitalization like this. How about:
```suggestion
            description: 'Please analyze the schema in the schema tab before sharing the schema.',
```

Reply:
Sounds good, updated!